### ISSUE TITLE
ServiceBinding for Kafka should not use localhost values if no bindings are present and fail fast when starting with invalid values

### ISSUE THREAD

**Issue by wtrocki** [quarkusio/quarkus#16682]
## Describe the bug

The typical approach for Kubernetes is to fail fast. If there is Kubernetes service binding plugin used proper behavior would 
be to look for the values and if they are not present fail application fast. 

This way application can stop etc. Currently app will try to connect to localhost (which is due to values being undefined).
My take is that I would love this to be consistent with behaviour without service binding plugin present (without kafka.* properties/profile/env app will fail to start)

(Note this is improvement and not blocking anything right now)

**Comment by wtrocki** [comment]
This is minor fix. I would love to contribute it based on agreement

**Comment by geoand** [comment]
Fine with me, but ultimately it's up to @cescoffier to decide the desired behavior

**Comment by cescoffier** [comment]
While I agree, it's not that simple. We need to look at the issue from a developer standpoint. Let's imagine a developer using Quarkus (great choice!), and starting to work on the application locally. It should work OOTB, and the broker may be running on the localhost. 

Now, our developer adds the service-binding extension because the targetted Kubernetes environment is configured to be used with service binding. Adding the extension is something the developer does on the local machine, so, should not prevent him to use a local broker, even if running the application using `java -jar ...`.

So, basically, we need to verify in the Kafka extension (at runtime) if we have the service binding extension but also if we are running on Kube and if the property is not defined somehow (env or something else overriding the service bindings). The extension does only create the configuration, so, how should it produce a failure (well, we can do a system exit or throwing a runtime exception and making sure this is run at bootstrap)?  Also, how can we detect we run on Kube? It's more specific than running in a container.

**Comment by cescoffier** [comment]
@geoand can we detect if service bindings are enabled? If so, we could detect in the extension that the `kafka.bootstrap.servers` is not set.

Note that this goes a bit against dev services, where we need to have `kafka.boostrap.servers` unset to get your broker started. So, it would mean that we can't use service bindings and dev services together in dev mode. The check proposed here would only happen in "prod".

**Comment by geoand** [comment]
@cescoffier yes, we can certainly do that

**Comment by geoand** [comment]
@cescoffier just to be clear, what you are looking for is the following

If:

* `quarkus-kubernetes-service-binding` is present
* `quarkus.kubernetes-service-binding.enabled=true`
* `quarkus-kafka-client` is present

Then we should fail if `kafka.bootstrap.servers` was not set by the service binding code?

**Comment by geoand** [comment]
In any case, I opened https://github.com/quarkusio/quarkus/pull/18066
